Source/WebKit/UIProcess/WebProcessProxy.cpp

```diff
@@ -2051,6 +2051,25 @@ void WebProcessProxy::logDiagnosticMessageForResourceLimitTermination(const Stri
     }
 }
 
+void WebProcessProxy::memoryPressureStatusChanged(SystemMemoryPressureStatus status)
+{
+    m_memoryPressureStatus = status;
+
+#if ENABLE(WEB_PROCESS_SUSPENSION_DELAY)
+    protectedProcessPool()->memoryPressureStatusChangedForProcess(Ref { *this }, status);
```

Comment:
`Ref { }` is unnecessary.

Reply:
Hm yeah, I guess the caller is supposed to already protect us on the stack, so I'll fix that.

However looking at this a bit more closely, I think I need to change this to something like:

```
if (RefPtr pool = m_processPool.get())
   pool->memoryPressureStatusChangedForProcess
```

Because in theory, while cached/prewarmed processes should be suspended, a low memory notification could sneak in just before they're suspended. If this function runs on a cached or suspended WebProcessProxy, then `protectedProcessPool()` crashes.